### PR TITLE
clientv3: balancer panics when call up after close

### LINKED ISSUE #6395: simpleBalancer.Up: close of closed channel
via semaphore (https://semaphoreci.com/coreos/etcd/branches/master/builds/908)

```
=== RUN   ExampleKV_putErrorHandling
panic: close of closed channel

goroutine 615 [running]:
panic(0x1041b60, 0xc823231f80)
	/usr/local/golang/1.6.3/go/src/runtime/panic.go:481 +0x3ff
github.com/coreos/etcd/clientv3.(*simpleBalancer).Up(0xc8232008c0, 0xc8201eeb07, 0x2d, 0x0, 0x0, 0x0)
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/clientv3/balancer.go:79 +0xa5
google.golang.org/grpc.(*addrConn).resetTransport(0xc8233e0000, 0x0, 0x0, 0x0)
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/google.golang.org/grpc/clientconn.go:694 +0x10f6
google.golang.org/grpc.(*ClientConn).resetAddrConn.func1(0xc8233e0000)
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/google.golang.org/grpc/clientconn.go:459 +0x38
created by google.golang.org/grpc.(*ClientConn).resetAddrConn
	/home/runner/workspace/src/github.com/coreos/etcd/gopath/src/google.golang.org/grpc/clientconn.go:468 +0xb9
```

### PR DESCRIPTION
**Pull request by xiang90**
Fix the issue by adding a simple guard variable.

Fix #6395 

/cc @heyitsanthony

### CODE REVIEW

File: clientv3/balancer.go [hunk 1]
```diff
@@ -128,13 +137,17 @@ func (b *simpleBalancer) Notify() <-chan []grpc.Address { return b.notifyCh }
 
 func (b *simpleBalancer) Close() error {
 	b.mu.Lock()
+	defer b.mu.Unlock()
+	if b.closed {
```
Comment (heyitsanthony) [inline clientv3/balancer.go]: grpc can call `Close` more than once too? if not, should probably omit the check?

File: clientv3/balancer.go [hunk 2]
```diff
@@ -74,15 +76,22 @@ func (b *simpleBalancer) ConnectNotify() <-chan struct{} {
 
 func (b *simpleBalancer) Up(addr grpc.Address) func(error) {
 	b.mu.Lock()
+	defer b.mu.Unlock()
+
+	if b.closed {
```
Comment (heyitsanthony) [inline clientv3/balancer.go]: comment about grpc being racey?

**Comment by heyitsanthony** [comment]
lgtm

File: clientv3/balancer.go (hunk 1 above)
Comment (xiang90) [inline clientv3/balancer.go]: i do not know. I would rather keep this to be safe.

File: clientv3/balancer.go (hunk 2 above)
Comment (xiang90) [inline clientv3/balancer.go]: sure.

File: clientv3/balancer.go (hunk 1 above)
Comment (heyitsanthony) [inline clientv3/balancer.go]: ok whatever, just add TODO about the uncertainty then?

Comment (xiang90) [inline clientv3/balancer.go]: sure.

**Comment by xiang90** [comment]
@heyitsanthony added comments. Will merge when CI turns green.

**Pull request merged**